flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/ResultPartitionTest.java

```diff
@@ -148,15 +148,17 @@ public void testAddOnReleasedBlockingPartition() throws Exception {
 	/**
 	 * Tests {@link ResultPartition#addBufferConsumer} on a partition which has already been released.
 	 *
-	 * @param pipelined the result partition type to set up
+	 * @param partitionType the result partition type to set up
 	 */
-	protected void testAddOnReleasedPartition(final ResultPartitionType pipelined)
+	protected void testAddOnReleasedPartition(final ResultPartitionType partitionType)
 		throws Exception {
 		BufferConsumer bufferConsumer = createFilledBufferConsumer(BufferBuilderTestUtils.BUFFER_SIZE);
 		ResultPartitionConsumableNotifier notifier = mock(ResultPartitionConsumableNotifier.class);
 		try {
-			ResultPartition partition = createPartition(notifier, pipelined, true);
-			partition.release();
+			NetworkResultPartition networkPartition = createNetworkResultPartition(partitionType);
+			ResultPartition partition = new ResultPartition(
```

Comment:
do we need `ResultPartition` in this test?
`networkPartition. addBufferConsumer` seems to work as well.

Reply:
I think we might still need the `ResultPartition`.
This test wants to verify the notification logic after releasing partition. The `partition.addBufferConsumer` would return false after released, then it would never trigger notification which is verified in finally path. And the notification logic is removed from `networkPartition`, so it can not be verified directly.

Reply:
agree